src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseResource.java

```diff
@@ -547,6 +550,26 @@ public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithAuxiliaryRe
         return ResponseEntity.ok(programmingExercise);
     }
 
+    /**
+     * GET /programming-exercises : Queries a programming exercise by its project key.
+     *
+     *
+     * @param projectKey the project key of the programming exercise
+     *
+     * @return the ProgrammingExercise of this project key in an ResponseEntity or 404 Not Found if no exercise exists
```

Comment:
The JavaDoc is incorrect, this endpoint returns much more data. We agreed that for new REST endpoints, we want to use DTOs and not entities anymore, because those can easily leak data